src/config/script/index.js

```diff
@@ -43,87 +65,85 @@ class Script {
   /**
    * Runs the script
    *
-   * @param string shell - the shell to run the command within
+   * @param {string} shell - the shell to run the command within
    */
   run(shell = DEFAULT_SHELL) {
-    return new Promise(() => {
-      let key = this.getName();
-
-      const subscriber = inquirerPrompts.subscribe(
-        ({ answer }) => {
-          switch (getUndocumentedChoice(answer)) {
-            case QUIT_COMMAND:
-              subscriber.unsubscribe();
-              safeExit();
-              break;
-            case BACK_COMMAND: {
-              key = getParentKey(key);
+    let key = this.getName();
+
+    const subscriber = inquirerPrompts.subscribe(
+      ({ answer }) => {
+        switch (getUndocumentedChoice(answer)) {
+          case QUIT_COMMAND:
+            subscriber.unsubscribe();
+            safeExit();
+            break;
+          case BACK_COMMAND: {
+            key = getParentKey(key);
+            let option = this.getOption(key);
+            if (!option.getMessage()) {
+              // Option didn't have a message; set the default message
+              option = Option.copy(option);
+              option.updateMessage('Choose an option');
+            }
+            prompts.next(option);
+            break;
+          }
+          default:
+            // eslint-disable-next-line no-param-reassign
+            answer = getUndocumentedChoice(answer);
+            if (endsWithWhitespace(answer)) {
+              // eslint-disable-next-line no-param-reassign
+              answer = replaceWhitespace(answer, KEY_SEPARATOR);
+            }
+            key = `${key}.${answer}`;
+            if (this.getOption(key)) {
               let option = this.getOption(key);
               if (!option.getMessage()) {
```

Comment:
This could be refactored into `getOptionMessageOrDefault(option)` or `option.getMessage()` could just return the default message if it has none. Option constructor could just set the default message if not set.